Step04/app/src/main/java/com/kwoolytech/step04/MainActivity.java

```diff
@@ -34,34 +35,35 @@ protected void onCreate(Bundle savedInstanceState) {
         ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawer, toolbar,
                 R.string.navigation_drawer_open, R.string.navigation_drawer_close);
 
-        drawer.setDrawerListener(toggle);
         toggle.syncState();
 
         NavigationView navigationView = (NavigationView)findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
 
-        dataModel = new HansWeatherDataModel();
+        dataModel   = new HansWeatherDataModel();
         httpClient  = new KwoolyHttp(MainActivity.this);
 
         initializeUi();
         initializeKwoolyHttpCallbackFunction();
 
-        queryWeatherJsonData();
+        queryWeatherJsonData(37.49, 127.01);
 
```

Comment:
I was stumped with the run-time permissioning stuff that were introduced from SDK 23. One that intrigued me was, we have many options when we implement the location related thing.
For this PR, I picked the simplest way, which is using a getLastKnown location. If there is no last known location, I chose not to request for current location because it is not worthy to consume my battery. ;)